foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_10_0/ItInfrastructureJavaUpgradeMigration.java

```diff
@@ -0,0 +1,42 @@
+package com.boozallen.aissemble.upgrade.migration.v1_10_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+
+public class ItInfrastructureJavaUpgradeMigration extends AbstractAissembleMigration {
+    private static final String OLD_JDK_VERSION = "openjdk:11-slim";
+    private static final String NEW_JDK_VERSION = "openjdk:17-jdk-slim";
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        try {
+            return (Files.readString(file.toPath()).contains(OLD_JDK_VERSION));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        try {
+            Files.writeString(file.toPath(), Files.readString(file.toPath())
+                    .replace(OLD_JDK_VERSION, NEW_JDK_VERSION));
```

Comment:
A: I can't immediately think of a realistic case where this would be a problem, but does it make sense to limit this to just `FROM` lines so we don't accidentally change too much?